bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java

```diff
@@ -158,8 +158,8 @@ protected <T> T getRequest(String path, Class<? extends T> dtoClass)
             throw new IndegoException(e);
         } catch (ExecutionException e) {
             Throwable cause = e.getCause();
-            if (cause != null && cause instanceof HttpResponseException) {
-                Response response = ((HttpResponseException) cause).getResponse();
+            if (cause != null && cause instanceof HttpResponseException exception) {
+                Response response = exception.getResponse();
```

Comment:
Since `exception` is as generic as `e`, it would be better to name according to type:
```suggestion
            if (cause != null && cause instanceof HttpResponseException httpResponseException) {
                Response response = httpResponseException.getResponse();
```